wdl-grammar/src/v1/lint/missing_runtime_block.rs

```diff
@@ -0,0 +1,270 @@
+//! workflow tasks _must_ have a `runtime` block to ensure portability.
+
+use std::collections::VecDeque;
+
+use nonempty::NonEmpty;
+use pest::iterators::Pair;
+use wdl_core::concern::code;
+use wdl_core::concern::lint;
+use wdl_core::concern::lint::Group;
+use wdl_core::concern::lint::Rule;
+use wdl_core::concern::Code;
+use wdl_core::file::Location;
+use wdl_core::Version;
+
+use crate::v1;
+
+/// Every task _must_ have a `runtime` block.
+#[derive(Debug)]
+pub struct MissingRuntimeBlock;
+
+impl<'a> MissingRuntimeBlock {
+    /// Creates an error corresponding to the task with a missing `runtime`
+    /// block.
+    fn missing_runtime_block(&self, location: Location) -> lint::Warning
+    where
+        Self: Rule<&'a Pair<'a, v1::Rule>>,
+    {
+        // SAFETY: this error is written so that it will always unwrap.
+        lint::warning::Builder::default()
+            .code(self.code())
+            .level(lint::Level::Medium)
```

Comment:
I chose `Medium` because it's not strictly required but probably won't work on any other machine unless it's a very simple POSIX compliant command.

Reply:
I think I stated this some time ago over Slack, but IMO "lint levels" don't make much sense. Taking a user's perspective, I don't see how to interact with the concept. What we've arbitrarily stated as important VS unimportant is unlikely to line up with a random external user's ideas.

I'd rather see "tags" added to each rule, such as "whitespace", "casing", "completeness", etc.
That seems more useful for a user to interact with. (Open to other ideas as well! Or maybe I don't understand "levels" 🤷 )

Of course that change doesn't belong in this PR, but my point is I don't have a strong opinion other than wanting this concept changed. So `Medium` is a-ok for me.

Reply:
I generally think of them as low = you should probably change this for clarity or style reasons but it doesn't really matter. Whereas high would be for something where you should really examine it to see if it is really what you mean to do (because it is probably wrong). For example in C `if (i = 0) { ... }` is correct but you almost certainly mean `if (i == 0) {...}`. I would make that high.

You could also use them in CI scripts. So "low" warnings are passable but "medium" or above will fail the build?

Reply:
Ok. Thank you for providing more concrete definitions. The concept isn't as strange to me anymore.
I have other thoughts here, but I don't think this thread is the place. `Medium` makes enough sense to me, don't want to hold up your PR over this :)

Reply:
Yeah, I generally agree that the concept is somewhat ill-defined but also warrants discussion elsewhere. @a-frantz can you make a discussion with your thoughts and ping @markjschreiber and I?